milli/src/update/index_documents/typed_chunk.rs

```diff
@@ -673,22 +673,14 @@ pub(crate) fn write_typed_chunk_into_index(
                 .get(&embedder_name)
                 .map_or(false, |conf| conf.2);
             // FIXME: allow customizing distance
-            let writers: Vec<_> = crate::vector::arroy_db_range_for_embedder(embedder_index)
-                .map(|k| ArroyWrapper::new(index.vector_arroy, k, binary_quantized))
-                .collect();
+            let writer = ArroyWrapper::new(index.vector_arroy, embedder_index, binary_quantized);
 
             // remove vectors for docids we want them removed
             let merger = remove_vectors_builder.build();
             let mut iter = merger.into_stream_merger_iter()?;
             while let Some((key, _)) = iter.next()? {
                 let docid = key.try_into().map(DocumentId::from_be_bytes).unwrap();
-
-                for writer in &writers {
-                    // Uses invariant: vectors are packed in the first writers.
-                    if !writer.del_item(wtxn, expected_dimension, docid)? {
-                        break;
-                    }
-                }
+                writer.del_item_raw(wtxn, expected_dimension, docid)?;
```

Comment:
Huuum, I would vastly prefer this approach because I didn’t really understand why what we were doing previously was working.

> What we want here (and were doing previously) is to remove all vectors for that docid.

But are you sure about that?
Because from what I see, the previous code was doing the same thing as me:
```rust
            let writers: Vec<_> = crate::vector::arroy_db_range_for_embedder(embedder_index)
                .map(|k| ArroyWrapper::new(index.vector_arroy, k, binary_quantized))
                .collect();
...
               for writer in &writers {
                    // Uses invariant: vectors are packed in the first writers.
                    if !writer.del_item(wtxn, expected_dimension, docid)? {
                        break;
                    }
                }
```

It goes through all the writers from and `break` when it finds something matching 🤔 

Or are you saying that we could get rids of the entire while loop?